### ISSUE TITLE
Lack of coherence in `sct_label_utils` regarding `RPI` orientation

### ISSUE THREAD

**Issue by NathanMolinier** [spinalcordtoolbox/spinalcordtoolbox#4083]
### Description

In the `api script` [labels.py](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py) some functions are applying reorientation steps to `RPI` before applying a process, however, this behavior is not present in most of the functions like: [add](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L31-L42), [create_labels_empty](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L45-L57), [create_labels](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L60-L72), [cubic_to_point](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L152-L190), [labelize_from_discs](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L219-L241), [check_missing_label](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L280-L306), [compute_mean_squared_error](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L310-L335), [remove_missing_labels](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L338-L352)...

For me, this reorientation check or manipulation should be applied in most of these cases like in [sct_create_mask.py](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/scripts/sct_create_mask.py) where the image orientation is changed before creating the mask. For example, in [create_labels_empty](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/labels.py#L45-L57) the creation of the new label is done without considering the orientation of the input.

**Comment by joshuacwnewton** [comment]
In some of these functions, I can imagine that reorientation is not necessary (e.g. `add`). 

But, you're right that for some functions, reorientation to RPI is missing. (For example, `create_labels` and `create_labels_empty` do not reorient, yet `create_labels_along_segmentation` _does_ reorient to RPI.)

Thank you for pointing out this inconsistency. It is definitely worth exploring further.

**Comment by NathanMolinier** [comment]
I agree that some functions could maybe avoid this reorientation step in my list. However, as I said, I have been using `sct_create_mask`and I felt strange about the different behaviour when trying to create a single label in an image using `sct_label_utils -create`.
